Sprint_6

AlekseyShterts opened the pull request:
1. Добвлены тесты на классы Lion, Cat, Feline.
2. Устранена зависимость Lion от Feline.

AlekseyShterts committed:
Try one.

1. Добвлены тесты на классы Lion, Cat, Feline.
2. Устранена зависимость Lion от Feline.

pomegranate282922 commented on pom.xml:
```diff
@@ -25,7 +25,62 @@
+            <artifactId>junit</artifactId>
+            <version>4.13.2</version>
+        </dependency>
+        <dependency>
```
Селениде? Зачем?

pomegranate282922 commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,50 @@
+    private Feline feline;
+
+    @Spy
+    private Feline felineSpy = new Feline();
```
Вот здесь инициализация НЕ нужна!

pomegranate282922 replied:
НЕ нужна!!!!!

pomegranate282922 commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,43 @@
+        boolean actual = lion.doesHaveMane();
+        assertEquals("Проверка пола",expected, actual);
+
+    }
```
А как же остальные методы? И нужно применить моки!

pomegranate282922 reviewed (commented):
Прошу ещё дописать тесты для класса Lion. 
и ещё хотелось бы увидеть тесты на исключения.

AlekseyShterts committed:
Try two.

1. dependency junit - используется для org.junit.runner.RunWith; import org.junit.runners.Parameterized; и тд.
2. Class LionTest - добавлены тесты по остальным методам + проверка исключения + добавлены моки.
3. Class FelineTest - удалена лишняя инициализация.

pomegranate282922 commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,77 @@
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {"Самка", false, "Используйте допустимые значения пола животного - самец или самка"},
```
Нужен ли третий параметр, если он всегда одинаковый?

pomegranate282922 commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @Test
+    public void unknownAnimalReturnException() {
```
Супер! Единственное, что этот тест у тебя повторится дважды, так как он находится в параметризованном классе.

pomegranate282922 reviewed (approved):
Зачёт!
